src/compiler/sys/logger/logger.ts

```diff
@@ -0,0 +1,75 @@
+import { Diagnostic } from '../../diagnostic';
+
+/**
+ * This sets the log level hierarchy for our terminal logger, ranging from
+ * most to least verbose.
+ *
+ * Ordering the levels like this lets us easily check whether we should log a
+ * message at a given time. For instance, if the log level is set to `'warn'`,
+ * then anything passed to the logger with level `'warn'` or `'error'` should
+ * be logged, but we should _not_ log anything with level `'info'` or `'debug'`.
+ *
+ * If we have a current log level `currentLevel` and a message with level
+ * `msgLevel` is passed to the logger, we can determine whether or not we should
+ * log it by checking if the log level on the message is further up or at the
+ * same level in the hierarchy than `currentLevel`, like so:
+ *
+ * ```ts
+ * LOG_LEVELS.indexOf(msgLevel) >= LOG_LEVELS.indexOf(currentLevel)
+ * ```
+ *
+ * NOTE: for the reasons described above, do not change the order of the entries
+ * in this array without good reason!
+ */
+export const LOG_LEVELS = ['debug', 'info', 'warn', 'error'] as const;
```

Comment:
I moved the stuff for logger into this module so that the sub-project this module is part of won't have to depend on the declaration files when they are eventually built as part of a separate project later on.

I tried (both here and with the `Diagnostic` change) to do two things:

- retain the same exports from `src/declarations` by re-exporting from their new location
- update all imports across the project to point at the new location rather than importing from `src/declarations`